src/Internal/ParameterHandler/Factory/ParameterHandlerFactoryProvider.php

```diff
@@ -0,0 +1,46 @@
+<?php
+declare(strict_types=1);
+
+namespace Retrofit\Internal\ParameterHandler\Factory;
+
+use Retrofit\Attribute\Body;
+use Retrofit\Attribute\Field;
+use Retrofit\Attribute\FieldMap;
+use Retrofit\Attribute\Header;
+use Retrofit\Attribute\HeaderMap;
+use Retrofit\Attribute\Part;
+use Retrofit\Attribute\PartMap;
+use Retrofit\Attribute\Path;
+use Retrofit\Attribute\Query;
+use Retrofit\Attribute\QueryMap;
+use Retrofit\Attribute\QueryName;
+use Retrofit\Attribute\Url;
+use Retrofit\Internal\ConverterProvider;
+
+class ParameterHandlerFactoryProvider
+{
+    private array $attributeNameToFactory;
+
+    public function __construct(ConverterProvider $converterProvider)
+    {
+        $this->attributeNameToFactory = [
```

Comment:
I'm not 100% sure about this code. Maybe it would be better to have static mapping:

```
Body::class => BodyParamterHandlerFactory::class
```

And change it into factory?

Reply:
Factory creates factory - weird to me. This class is created only one in the proxing (interface implementing) phase.

What are you worried about?